test/e2e/app-dir/interpolability-with-pages/navigation.test.ts

```diff
@@ -0,0 +1,68 @@
+import { createNext, FileRef } from 'e2e-utils'
+import { NextInstance } from 'test/lib/next-modes/base'
+import webdriver from 'next-webdriver'
+import { waitFor } from 'next-test-utils'
+
+describe('navigation between pages and app dir', () => {
+  let next: NextInstance
+
+  beforeAll(async () => {
+    next = await createNext({
+      files: new FileRef(__dirname),
+      dependencies: {
+        react: 'latest',
+        'react-dom': 'latest',
+        typescript: 'latest',
+        '@types/react': 'latest',
+        '@types/node': 'latest',
+      },
+    })
+  })
+  afterAll(() => next.destroy())
+
+  it('It should be able to navigate app -> pages', async () => {
+    const browser = await webdriver(next.url, '/app')
+    expect(await browser.elementById('app-page').text()).toBe('App Page')
+    browser.elementById('link-to-pages').click()
+    await waitFor(200)
```

Comment:
Can you replace all cases of `waitFor` with `browser.waitForElementByCss()` so that you no longer need to wait an arbitrary time which causes test flakes <picture data-single-emoji=":thumbs:" title=":thumbs:"><img class="emoji" width="20" height="auto" src="https://single-emoji.vercel.app/api/emoji/eyJhbGciOiJkaXIiLCJlbmMiOiJBMjU2R0NNIn0..btU5TeewbBYCz9UD.6pweEJ_WEf-2LmScis_SjydqDbzlg3K6nbLR3S-lOm61e6IiykJCdoNYB8PTemrRTYd_iDlVzNYQhqt685GoirldN5-qf2mXUTFvQA-UoNctbFG3xau9Zw.XiQzt0oNbFJZ43F9w5AFVw" alt=":thumbs:" align="absmiddle"></picture>

Reply:
Hmm, I tried. It's comitted. But it doesn't work properly. This fails:
```ts
    await browser.waitForElementByCss('#pages-page')
    expect(await browser.hasElementByCssSelector('#pages-page')).toBeTrue()
```

Reply:
Oh I need to `await` it. The types are wrong. Will fix it.

Reply:
So fixing it will need a small refactor. Will do in https://linear.app/vercel/issue/NEXT-290/split-waitforelementbycss-into-two-functions-for-correct-typing